res/res_pjsip_sdp_rtp.c

```diff
@@ -1528,7 +1528,7 @@ static void set_session_media_remotely_held(struct ast_sip_session_media *sessio
 			((addrs != NULL) && ast_sockaddr_is_any(addrs)) ||
 			pjmedia_sdp_media_find_attr2(media, "sendonly", NULL) ||
 			pjmedia_sdp_media_find_attr2(media, "inactive", NULL)) {
-			if (!session_media->remotely_held) {
+			if (!session_media->remotely_held && !session->endpoint->suppress_moh_on_remote_hold) {
 				session_media->remotely_held = 1;
```

Comment:
If we don't set remotely_held, what occurs if we send a re-INVITE? Do we try to set it back to sendrecv?

Reply:
@jcolp Oops, I didn't see this earlier.

Here's what I tested...
Alice has suppress_moh_on_sendonly set to no.
Bob has suppress_moh_on_sendonly set no yes.
Alice calls Bob.
Bob answers then places call on hold by sending re-INVITE with sendonly.
Asterisk responds with 200 OK with recvonly.
Nothing happens to Alice's media session and no MOH is played.
Alice places call on hold by sending re-INVITE with sendonly.
Asterisk responds with 200 OK with recvonly.
Nothing happens to Bob's media session. He's still in sendonly/recvonly.

Now both are in sendonly/recvonly.

If Alice resumes with sendrecv, she hears nothing.
If Alice stays in sendonly/recvonly and Bob resumes with sendrecv, he hears MOH because it's not suppressed on Alice's endpoint.
When both resume with sendrecv, two-way audio is resumed.

If you have a specific scenario in mind let me know and I'll test it.

Reply:
I was more thinking about Asterisk generated re-INVITE, for example if a connected line update is done.

Reply:
Just to be sure nothing else is affected, I've moved the check for suppress_moh_on_sendonly down to the apply_negotiated_sdp_stream() function so the only thing it does is suppress the MOH without disturbing the remotely_held flag...

```c
if (session_media->remotely_held) {
	/* The remote side has put us on hold */
	if (!session->endpoint->suppress_moh_on_sendonly) {
		ast_queue_hold(session->channel, session->endpoint->mohsuggest);
		ast_rtp_instance_stop(session_media->rtp);
		ast_queue_frame(session->channel, &ast_null_frame);
	}
	session_media->remotely_held_changed = 0;
} else {
	/* The remote side has taken us off hold */
	if (!session->endpoint->suppress_moh_on_sendonly) {
		ast_queue_unhold(session->channel);
		ast_queue_frame(session->channel, &ast_null_frame);
	}
	session_media->remotely_held_changed = 0;
}
```
I originally had the check here but moved it up to set_session_media_remotely_held() because I thought other things triggered by remotely_held needed to be suppressed.  When I realized that wasn't the case, I meant to move it back to apply_negotiated_sdp_stream() but forgot.

I'll push the change up when I'm done testing.

Reply:
> I was more thinking about Asterisk generated re-INVITE, for example if a connected line update is done.

Ah connected line.  I can test that easily.

Reply:
Yep, was an issue with the original PR.  The session got renegotiated back to sendrecv.
With the update above, the re-INVITE goes back to Bob as recvonly.
PR all better now.